qiskit_machine_learning/neural_networks/effective_dimension.py

```diff
@@ -0,0 +1,354 @@
+
+    def num_weights(self) -> int:
+        """Returns the dimension of the model according to the definition from [1]."""
+        return self._model.num_weights
+
+    @property
+    def weight_samples(self) -> np.ndarray:
+        """Returns network parameters (weights)."""
+        return self._weight_samples
+
+    @weight_samples.setter
+    def weight_samples(self, weight_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network parameters (weights)."""
+        if isinstance(weight_samples, int):
+            # random sampling from uniform distribution
+            self._weight_samples = algorithm_globals.random.uniform(
+                0, 1, size=(weight_samples, self._model.num_weights)
+            )
+        else:
+            weight_samples = np.asarray(weight_samples)
+            if (
+                len(weight_samples.shape) > 2
+                or weight_samples.shape[1] is not self._model.num_weights
+            ):
+                raise QiskitMachineLearningError(
+                    f"The Effective Dimension class expects"
+                    f" a parameter array of shape (M, qnn.num_weights)."
+                    f" Got {weight_samples.shape}."
+                )
+            self._weight_samples = weight_samples
+
+        self._num_weight_samples = len(self._weight_samples)
+
+    @property
+    def input_samples(self) -> np.ndarray:
+        """Returns network input samples."""
+        return self._input_samples
+
+    @input_samples.setter
+    def input_samples(self, input_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network input samples."""
+        if isinstance(input_samples, int):
+            # random sampling from normal distribution
+            self._input_samples = algorithm_globals.random.normal(
+                0, 1, size=(input_samples, self._model.num_inputs)
+            )
+        else:
+            input_samples = np.asarray(input_samples)
+            if len(input_samples.shape) > 2 or input_samples.shape[1] is not self._model.num_inputs:
+                raise QiskitMachineLearningError(
+                    f"The Effective Dimension class expects"
+                    f" an input sample array of shape (N, qnn.num_inputs)."
+                    f" Got {input_samples.shape}."
+                )
+            self._input_samples = input_samples
+
+        self._num_input_samples = len(self._input_samples)
+
+    def run_monte_carlo(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        This method computes the model's Monte Carlo sampling for a set of input samples and
+        parameters (weights).
+
+        Returns:
+             grads: QNN gradient vector, result of backward passes, of shape
+                ``(num_samples * num_params, output_size, num_weights)``.
+             outputs: QNN output vector, result of forward passes, of shape
+                ``(num_samples * num_params, output_size)``.
+        """
```

Comment:
This is still referring to `parameters` and `num_samples`, `num_params`

Reply:
Updated